graphene/src/document.rs

```diff
@@ -375,6 +375,23 @@ impl Document {
 		Ok(())
 	}
 
+	fn mark_layer_type_as_dirty(root: &mut Layer, data_type: LayerDataTypeDiscriminant) -> bool {
+		if LayerDataTypeDiscriminant::from(&root.data) == data_type {
+			root.cache_dirty = true;
+		} else if let LayerDataType::Folder(f) = &mut root.data {
+			let mut dirty = false;
+			for l in f.layers_mut() {
+				dirty = Self::mark_layer_type_as_dirty(l, data_type) || dirty;
+			}
+			root.cache_dirty = dirty;
+		}
+		root.cache_dirty
+	}
+
+	pub fn mark_all_layer_type_as_dirty(&mut self, data_type: LayerDataTypeDiscriminant) -> bool {
```

Comment:
"type" should probably be plural: "types". Also could you please comment these two functions? (Or at least the non-plural one.)